contracts/vesting/VestingEscrowFactory.vy

```diff
@@ -24,80 +38,156 @@ interface VestingEscrowSimple:
     @param target `VestingEscrowSimple` contract address
+    @param yfi YFI token address
+    @param owner The owner of the factory that can add/remove liquid lockers and their operators
     """
     TARGET = target
+    YFI = ERC20(yfi)
+    OWNER = owner
 
 @external
-def deploy_vesting_contract(
-    token: ERC20,
-    recipient: address,
-    amount: uint256,
-    vesting_duration: uint256,
+def create_vest(
+    recipient: address, 
+    amount: uint256, 
+    vesting_duration: uint256, 
     vesting_start: uint256 = block.timestamp,
-    cliff_length: uint256 = 0,
-    open_claim: bool = True,
-    owner: address = msg.sender,
-) -> address:
+    cliff_length: uint256 = 0
+) -> uint256:
     """
-    @notice Deploy a new vesting contract
-    @dev Prior to deployment you must approve `amount` tokens
-    @param token ERC20 token being distributed
+    @notice Create a new YFI vest
+    @dev Prior to deployment you must approve `amount` YFI
     @param recipient Address to vest tokens for
     @param amount Amount of tokens being vested for `recipient`
     @param vesting_duration Time period (in seconds) over which tokens are released
     @param vesting_start Epoch time when tokens begin to vest
-    @param open_claim Switch if anyone can claim for `recipient`
-    @param owner Vesting contract owner
+    @param cliff_length Duration (in seconds) after which the first portion vests
+    @return Vest index
     """
     assert cliff_length <= vesting_duration  # dev: incorrect vesting cliff
     assert vesting_start + vesting_duration > block.timestamp  # dev: just use a transfer, dummy
     assert vesting_duration > 0  # dev: duration must be > 0
-    assert recipient not in [self, empty(address), token.address, owner] # dev: wrong recipient
+    assert recipient not in [self, empty(address), YFI.address, OWNER] # dev: wrong recipient
+    assert amount > 0
 
-    escrow: address = create_minimal_proxy_to(TARGET)
-    VestingEscrowSimple(escrow).initialize(
-        owner,
-        token,
+    idx: uint256 = self.num_vests
+    self.num_vests = idx + 1
+    self.pending_vests[idx] = Vest({
+        recipient: recipient,
+        amount: amount,
+        duration: vesting_duration,
+        start: vesting_start,
+        cliff: cliff_length
+    })
+    assert YFI.transferFrom(msg.sender, self, amount, default_return_value=True)
+
+    log VestingEscrowCreated(
+        msg.sender,
         recipient,
+        idx,
         amount,
         vesting_start,
-        vesting_start + vesting_duration,
-        cliff_length,
-        open_claim,
+        vesting_duration,
+        cliff_length
     )
 
-    # skip transferFrom and approve and send directly to escrow
-    assert token.transferFrom(msg.sender, escrow, amount, default_return_value=True)  # dev: funding failed
+    return idx
 
-    log VestingEscrowCreated(
+@external
+def deploy_vesting_contract(
+    idx: uint256,
+    token: address,
+    amount: uint256,
```

Comment:
do we want to check if amount is the exact amount of vesting, or do we want to give options to split liquid lockers for vesting?

Reply:
Users have the option to deploy to multiple lockers if they wish

Reply:
Users have the choice to split their vest between multiple lockers.